Re-review API validation for CSI Inline Migration

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
While glancing through storage validation.go, I noticed that we were doing feature gate checking in validation for CSI inline migration.  At first glance, I think this will cause validation errors when downgrading to a version with the feature disabled.

https://github.com/kubernetes/kubernetes/commit/c34309acdf1813961051023e913e07ec7a535428#diff-5aba802af469dbe928456a4b2c7988adR177

**What you expected to happen**:
We generally don't do feature gate checking in validation in order to support downgrade cases.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

In validateVolumeAttachmentSource :
```
	case source.InlineVolumeSpec != nil:
		if utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) {
			allErrs = append(allErrs, storagevalidation.ValidatePersistentVolumeSpec(source.InlineVolumeSpec, "", true, fldPath.Child("inlineVolumeSpec"))...)
		} else {
			allErrs = append(allErrs, field.Forbidden(fldPath, "may not specify inlineVolumeSpec when CSIMigration feature is disabled"))
		}
	}
```
If we don't check feature gate, it seems source.InlineVolumeSpec != nil case can be dropped.
Otherwise:
```
--- FAIL: TestVolumeAttachmentValidation (0.00s)
    validation_test.go:231: expected success: {{ } {foo-with-inlinespec      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {myattacher {<nil> 0x21fcb80} mynode} {false map[] <nil> <nil>}} [spec.source: Forbidden: may not specify inlineVolumeSpec when CSIMigration feature is disabled]
    validation_test.go:231: expected success: {{ } {foo-with-inlinespec-and-status      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {myattacher {<nil> 0x21fcb80} mynode} {true map[foo:bar] 0xc0001c2ff0 0xc0001c3020}} [spec.source: Forbidden: may not specify inlineVolumeSpec when CSIMigration feature is disabled]
```

--- a/pkg/apis/storage/validation/validation.go
+++ b/pkg/apis/storage/validation/validation.go
@@ -187,11 +187,7 @@ func validateVolumeAttachmentSource(source *storage.VolumeAttachmentSource, fldP
 			allErrs = append(allErrs, field.Required(fldPath.Child("persistentVolumeName"), "must specify non empty persistentVolumeName"))
 		}
 	case source.InlineVolumeSpec != nil:
-		if utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) {
-			allErrs = append(allErrs, storagevalidation.ValidatePersistentVolumeSpec(source.InlineVolumeSpec, "", true, fldPath.Child("inlineVolumeSpec"))...)
-		} else {
-			allErrs = append(allErrs, field.Forbidden(fldPath, "may not specify inlineVolumeSpec when CSIMigration feature is disabled"))
-		}
+		allErrs = append(allErrs, storagevalidation.ValidatePersistentVolumeSpec(source.InlineVolumeSpec, "", true, fldPath.Child("inlineVolumeSpec"))...)
 	}
 	return allErrs
 }

--- a/pkg/apis/storage/validation/validation_test.go
+++ b/pkg/apis/storage/validation/validation_test.go
@@ -394,14 +394,6 @@ func TestVolumeAttachmentValidation(t *testing.T) {
 				},
 			},
 		},
-	}
-	for _, volumeAttachment := range migrationDisabledSuccessCases {
-		if errs := ValidateVolumeAttachment(&volumeAttachment); len(errs) != 0 {
-			t.Errorf("expected success: %v %v", volumeAttachment, errs)
-		}
-	}
-
-	migrationDisabledErrorCases := []storage.VolumeAttachment{
 		{
 			// InlineSpec specified with migration disabled
 			ObjectMeta: metav1.ObjectMeta{Name: "foo"},
@@ -414,13 +406,11 @@ func TestVolumeAttachmentValidation(t *testing.T) {
 			},
 		},
 	}
-
-	for _, volumeAttachment := range migrationDisabledErrorCases {
-		if errs := ValidateVolumeAttachment(&volumeAttachment); len(errs) == 0 {
-			t.Errorf("expected failure: %v %v", volumeAttachment, errs)
+	for _, volumeAttachment := range migrationDisabledSuccessCases {
+		if errs := ValidateVolumeAttachment(&volumeAttachment); len(errs) != 0 {
+			t.Errorf("expected success: %v %v", volumeAttachment, errs)
 		}
 	}
-
 }
 
 func TestVolumeAttachmentUpdateValidation(t *testing.T) {

